backend/modules/pdf_processing/ai_extraction_service.py

```
@@ -33,6 +33,74 @@ class AIExtractionService:
     def __init__(self):
         self.api_key = OPENAI_API_KEY
         self.client = openai.OpenAI(api_key=self.api_key) if self.api_key else None
+
+    def _get_message_content(self, response: Any, context: str, fallback: Optional[str] = None) -> str:
+        """Safely extract OpenAI message content."""
+        try:
+            choice = response.choices[0]
+            message = choice.message
+        except Exception as exc:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)